signalfx/resource_signalfx_detector.go

```diff
@@ -321,6 +327,14 @@ func getPayloadDetector(d *schema.ResourceData) (*detector.CreateUpdateDetectorR
 	maxDelay := int32(d.Get("max_delay").(int) * 1000)
 	minDelay := int32(d.Get("min_delay").(int) * 1000)
 
+	var tags []string
+	if val, ok := d.GetOk("tags"); ok {
+		tags := []string{}
```

Comment:
This shadows the external `tags` variable, so tags are never actually set. See https://github.com/splunk-terraform/terraform-provider-signalfx/pull/306#issuecomment-870415564